src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -408,7 +410,7 @@ internal ImportCertificateResult ImportCertificate(string certificatePath, strin
             return ImportCertificateResult.Succeeded;
         }
 
-        public void CleanupHttpsCertificates()
+        public void CleanupHttpsCertificates(IReporter? reporter = null, bool isInteractive = false)
```

Comment:
`reporter` is added to provide feedback to the user.
`isInteractive` tracks whether we can prompt the user.

Reply:
You can't use the reporter here. This code is shared in multiple places and we don't want to bring CLI dependencies here. We use ETW to write event logs for tracing purposes.